Tests for PicketLink Federation subsystem

zebrik opened the pull request.

zebrik committed:
Possibility to mention test specific tracked issues in ResourceVerifier methods.

zebrik committed:
Add method to get subsystem configuration FinderNavigation.

zebrik committed:
Add abstration to click save button in WindowFragment.

zebrik committed:
Add methods to verify resource existence with timeout.

zebrik committed:
Tests for PicketLink Federation subsystem.

zebrik commented:
@rhatlapa please review

rhatlapa commented on basic/src/test/java/org/jboss/hal/testsuite/test/configuration/picketlink/PicketlinkOperations.java:
```diff
@@ -0,0 +1,65 @@
+            adminOps.reloadIfRequired();
+        }
+    }
+}
```
From API point of view, I would prefer having `reloadImmediatellyIfRequired` set as part of constructor. From your usage I see that you are using `enableExtension(false)` and `disableExtension(true)`, which would mean that you would need to initialize two `PicketlinkOperations` object => which is fine from my point of view and more clear when reading the code.

rhatlapa commented on basic/src/test/java/org/jboss/hal/testsuite/test/configuration/picketlink/federation/AbstractFederationTestCase.java:
```diff
@@ -0,0 +1,110 @@
+     * @return '{@code true}' unless system property '{@code federation.already.enabled}' is set to '{@code true}'
+     */
+    private static boolean changeSubsystem() {
+        return !Boolean.parseBoolean(ConfigUtils.get("federation.already.enabled"));
```
Aren't you missing default?

rhatlapa replied:
Looking to javadoc, it should be OK as it is.

rhatlapa commented on common/src/main/java/org/jboss/hal/testsuite/fragment/WindowFragment.java:
```diff
@@ -110,4 +117,13 @@ public String getTitle() {
+        } catch (WebDriverException e) {
+            log.debug("Button with label \"" + label + "\" not found");
+            throw e;
+        }
```
from method name I would guess that in case of fail it will only log in case of failure. But exception is thrown as well => I suggest to use more clear name of the method.

rhatlapa commented on basic/src/test/java/org/jboss/hal/testsuite/test/configuration/picketlink/PicketlinkOperations.java:
```diff
@@ -0,0 +1,65 @@
+/**
+ * Created by pjelinek on Nov 16, 2015
+ */
+public class PicketlinkOperations {
```
Add javadoc.

rhatlapa commented:
There needs to be only done some minor improvements regarding API and javadoc, otherwise looks good.

zebrik committed:
API cleanup

Removing some newlines to make code easier to read.

Make PicketLink CLI operations API more clean and documented.

Removing inesential method clickButtonAndLogIfFails

zebrik committed:
Add configurable default timeout to all ResourceVerifier methods.

zebrik commented:
@rhatlapa can you please review again?

zebrik commented:
I have changed last 2 commits. @rhatlapa can you please review again?

rhatlapa commented:
LGTM, do some minor squashing, then I will check it once more and merge.

zebrik commented:
Squashed a bit.

rhatlapa commented on basic/src/test/java/org/jboss/hal/testsuite/test/configuration/picketlink/federation/SpAttrTestCase.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Test
+    public void url() throws Exception {
+        edit(InputType.TEXT, spAddress, URL, new ModelNode("")).andVerifyFailure();
```
You are trying to detect whether the URL won't be an empty String, shouldn't this check be rather against undefined value?

zebrik committed:
UtilFedInputChecker API cleanup.

zebrik commented:
@rhatlapa please review again ;-)

zebrik commented:
Reload if required removed from `org.jboss.hal.testsuite.test.configuration.picketlink.federation.UtilFedInputChecker.edit` since it was a bit paranoid and moreover tested model not web console.
@rhatlapa please review for the last time.

rhatlapa commented:
LGTM, thanks for this huge update, merging.
